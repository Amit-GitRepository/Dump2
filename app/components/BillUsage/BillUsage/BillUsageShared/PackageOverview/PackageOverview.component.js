import Proptypes from 'prop-types';
import React, {Component} from 'react';
import styles from './PackageOverview.style';
import {ISText} from '../../../../Shared';
import {View} from 'react-native';

class PackageOverview extends Component {
  render () {
    const {title, subTitle, showSeparator} = this.props;
    return (
      <View style = {styles.planDetailsContainer}>
        <ISText type = 'BOLD' style = {styles.textTitle}>{title}</ISText>
        {subTitle ? <ISText style = {styles.textSubtitle}>{subTitle}</ISText> : null}
        {showSeparator ? <View style = {styles.bottomSeparator}/> : null}
      </View>
    );
  }
}
PackageOverview.defaultProps = {
  title: '',
  subTitle: '',
  showSeparator: true
};
PackageOverview.propTypes = {
  title: Proptypes.string,
  subTitle: Proptypes.string,
  showSeparator: Proptypes.bool
};
export default PackageOverview;
