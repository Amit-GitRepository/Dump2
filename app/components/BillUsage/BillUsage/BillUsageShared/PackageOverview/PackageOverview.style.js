import {colors, fonts} from '../../../../../themes/constants.styles';

export default {
  textTitle: {
    color: colors.SECONDARY_BG_TEXT_CONTRAST,
    fontSize: fonts.FONT_SIZE_NORMAL,
    textAlign: 'center'
  },
  textSubtitle: {
    textAlign: 'center',
    fontSize: fonts.FONT_SIZE_SMALL,
    color: colors.PRIMARY_SUBTEXT
  },
  planDetailsContainer: {
    paddingTop: 2,
    paddingHorizontal: 5
  },
  bottomSeparator: {
    borderBottomWidth: 1,
    marginTop: 10,
    borderBottomColor: colors.PRIMARY_BG_SEPARATOR
  }
};
