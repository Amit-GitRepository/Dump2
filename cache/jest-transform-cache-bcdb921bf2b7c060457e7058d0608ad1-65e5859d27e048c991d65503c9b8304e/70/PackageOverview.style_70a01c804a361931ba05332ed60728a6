Object.defineProperty(exports,"__esModule",{value:true});var cov_21n1dtdhew=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/BillUsageShared/PackageOverview/PackageOverview.style.js',hash='cf6426ac17ff9a2478c1a19b0fa6589dfd9070f7',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/BillUsageShared/PackageOverview/PackageOverview.style.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _constants=require('../../../../../themes/constants.styles');exports.default=

{
textTitle:{
color:_constants.colors.SECONDARY_BG_TEXT_CONTRAST,
fontSize:_constants.fonts.FONT_SIZE_NORMAL,
textAlign:'center'},

textSubtitle:{
textAlign:'center',
fontSize:_constants.fonts.FONT_SIZE_SMALL,
color:_constants.colors.PRIMARY_SUBTEXT},

planDetailsContainer:{
paddingTop:2,
paddingHorizontal:5},

bottomSeparator:{
borderBottomWidth:1,
marginTop:10,
borderBottomColor:_constants.colors.PRIMARY_BG_SEPARATOR}};