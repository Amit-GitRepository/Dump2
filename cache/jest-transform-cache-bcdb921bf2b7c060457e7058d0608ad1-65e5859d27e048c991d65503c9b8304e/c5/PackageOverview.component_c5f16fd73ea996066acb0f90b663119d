Object.defineProperty(exports,"__esModule",{value:true});var cov_k3oyuwdux=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/BillUsageShared/PackageOverview/PackageOverview.component.js',hash='43d3a3d4122f1a78a13b644ad50aa5e7eba37f81',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/BillUsageShared/PackageOverview/PackageOverview.component.js',statementMap:{'0':{start:{line:9,column:30},end:{line:9,column:40}},'1':{start:{line:10,column:4},end:{line:16,column:6}},'2':{start:{line:19,column:0},end:{line:22,column:2}},'3':{start:{line:23,column:0},end:{line:26,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:8,column:2},end:{line:8,column:3}},loc:{start:{line:8,column:12},end:{line:17,column:3}},line:8}},branchMap:{'0':{loc:{start:{line:13,column:9},end:{line:13,column:84}},type:'cond-expr',locations:[{start:{line:13,column:20},end:{line:13,column:77}},{start:{line:13,column:80},end:{line:13,column:84}}],line:13}},s:{'0':0,'1':0,'2':0,'3':0},f:{'0':0},b:{'0':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _PackageOverview=require('./PackageOverview.style');var _PackageOverview2=_interopRequireDefault(_PackageOverview);
var _Shared=require('../../../../Shared');
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

PackageOverview=function(_Component){_inherits(PackageOverview,_Component);function PackageOverview(){_classCallCheck(this,PackageOverview);return _possibleConstructorReturn(this,(PackageOverview.__proto__||Object.getPrototypeOf(PackageOverview)).apply(this,arguments));}_createClass(PackageOverview,[{key:'render',value:function render()
{cov_k3oyuwdux.f[0]++;var _ref=(cov_k3oyuwdux.s[0]++,
this.props),title=_ref.title,subTitle=_ref.subTitle;cov_k3oyuwdux.s[1]++;
return(
_react2.default.createElement(_reactNative.View,{style:_PackageOverview2.default.planDetailsContainer},
_react2.default.createElement(_Shared.ISText,{type:'BOLD',style:_PackageOverview2.default.textTitle},title),
subTitle?(cov_k3oyuwdux.b[0][0]++,_react2.default.createElement(_Shared.ISText,{style:_PackageOverview2.default.textSubtitle},subTitle)):(cov_k3oyuwdux.b[0][1]++,null),
_react2.default.createElement(_reactNative.View,{style:_PackageOverview2.default.bottomSeparator})));


}}]);return PackageOverview;}(_react.Component);cov_k3oyuwdux.s[2]++;

PackageOverview.defaultProps={
title:'',
subTitle:''};cov_k3oyuwdux.s[3]++;

PackageOverview.propTypes={
title:_propTypes2.default.string,
subTitle:_propTypes2.default.string};exports.default=

PackageOverview;