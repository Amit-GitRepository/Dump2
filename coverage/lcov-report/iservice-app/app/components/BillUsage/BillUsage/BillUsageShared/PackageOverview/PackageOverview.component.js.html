<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for iservice-app/app/components/BillUsage/BillUsage/BillUsageShared/PackageOverview/PackageOverview.component.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../../../index.html">All files</a> / <a href="index.html">iservice-app/app/components/BillUsage/BillUsage/BillUsageShared/PackageOverview</a> PackageOverview.component.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">43x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">43x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import Proptypes from 'prop-types';
import React, {Component} from 'react';
import styles from './PackageOverview.style';
import {ISText} from '../../../../Shared';
import {View} from 'react-native';
&nbsp;
class PackageOverview extends Component {
  render () {
    const {title, subTitle} = this.props;
    return (
      &lt;View style = {styles.planDetailsContainer}&gt;
        &lt;ISText type = 'BOLD' style = {styles.textTitle}&gt;{title}&lt;/ISText&gt;
        {subTitle ? &lt;ISText style = {styles.textSubtitle}&gt;{subTitle}&lt;/ISText&gt; : null}
        &lt;View style = {styles.bottomSeparator}/&gt;
      &lt;/View&gt;
    );
  }
}
PackageOverview.defaultProps = {
  title: '',
  subTitle: ''
};
PackageOverview.propTypes = {
  title: Proptypes.string,
  subTitle: Proptypes.string
};
export default PackageOverview;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 03 2017 16:14:06 GMT+0530 (IST)
</div>
</div>
<script src="../../../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../../../sorter.js"></script>
</body>
</html>
