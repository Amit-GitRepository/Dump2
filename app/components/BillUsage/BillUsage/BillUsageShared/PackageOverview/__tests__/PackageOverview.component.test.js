import PackageOverview from '../PackageOverview.component';
import React from 'react';
import renderer from 'react-test-renderer';

describe('PackageOverview component', () => {
  it('renders correctly', () => {
    const mockTitle = 'title';
    const mocksubTitle = 'sub title';
    const component = renderer.create(<PackageOverview title = {mockTitle} subTitle = {mocksubTitle}/>);
    expect(component).toMatchSnapshot();
  });
  it('renders correctly without props', () => {
    const component = renderer.create(<PackageOverview/>);
    expect(component).toMatchSnapshot();
  });
});