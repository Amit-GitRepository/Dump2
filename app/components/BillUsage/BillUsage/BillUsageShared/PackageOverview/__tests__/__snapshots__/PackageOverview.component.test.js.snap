// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PackageOverview component renders correctly 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 5,
      "paddingTop": 2,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#000000",
        "fontFamily": "SukhumvitSet-Bold",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    title
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#68747E",
        "fontFamily": "SukhumvitSet-Text",
        "fontSize": 14,
        "textAlign": "center",
      }
    }
  >
    sub title
  </Text>
  <View
    style={
      Object {
        "borderBottomColor": "#E9E9EA",
        "borderBottomWidth": 1,
        "marginTop": 10,
      }
    }
  />
</View>
`;

exports[`PackageOverview component renders correctly without props 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 5,
      "paddingTop": 2,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#000000",
        "fontFamily": "SukhumvitSet-Bold",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    
  </Text>
  <View
    style={
      Object {
        "borderBottomColor": "#E9E9EA",
        "borderBottomWidth": 1,
        "marginTop": 10,
      }
    }
  />
</View>
`;
